### PR TITLE
Docs: replace guide link in README.md

### LINKED ISSUE #1041: [Doc] In the README.md, developer guide link can not be accessed
**What happened**:

In the README.md, developer guide link can not be accessed
developer [guide](https://github.com/openyurtio/openyurt/blob/master/docs/developer-guide.md) 

`./docs/developer-guide.md`
`https://github.com/openyurtio/openyurt/blob/master/docs/developer-guide.md`

I found the developer-guide.md  in a version before, but the current master branch does not have this file. Should I delete this link?

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- OpenYurt version:
- Kubernetes version (use `kubectl version`):
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

**others**

### PR DESCRIPTION
**Pull request by Lan-ce-lot**
Signed-off-by: Lancelot <1984737645@qq.com>

/kind documentation
#### What this PR does / why we need it:
In the README.md, developer guide link can not be accessed.And current master branch does not have developer-guide.md.

#### Which issue(s) this PR fixes:

Fixes #1041 

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

**Comment by Congrool** [comment]
Sounds good, but how about using https://openyurt.io/docs/developer-manuals/how-to-build-and-test instead?

File: README.md [hunk 1]
```diff
@@ -88,7 +88,7 @@ To experience the power of OpenYurt, please try the detailed [tutorials](https:/
 ### Contributing
 
 If you are willing to be a contributor for the OpenYurt project, please refer to our [CONTRIBUTING](CONTRIBUTING.md) document for details.
-We have also prepared a developer [guide](./docs/developer-guide.md) to help the code contributors.
+We have also prepared a developer guide to help the code contributors.
```
Comment (rambohe-ch) [inline README.md:91]: developer guide tutorials are removed to here: https://openyurt.io/docs/developer-manuals/how-to-contribute
please add the link for it.

**Comment by rambohe-ch** [comment]
> Sounds good, but how about using https://openyurt.io/docs/developer-manuals/how-to-build-and-test instead?

@Congrool maybe we need to add a link for `Developer Manuals` tab and reference it. of course, we can handle it in another pull request.

**Comment by Congrool** [comment]
> maybe we need to add a link for Developer Manuals tab and reference it. of course, we can handle it in another pull request.

Good idea

**Comment by Congrool** [comment]
@Lan-ce-lot Could you please squash your pr into 1 commits?

**Comment by Lan-ce-lot** [comment]
> @Lan-ce-lot Could you please squash your pr into 1 commit? You can use `git rebase -i` with squash word or `git reset && git commit` to do this.

Ok.

**Review by Congrool — COMMENTED** [review]
/lgtm

**Comment by Congrool** [comment]
LGTM, thanks for your contribution.

**Review by rambohe-ch — APPROVED** [review]

**Pull request merged**